stdlib/fun.mli

```diff
@@ -20,19 +20,42 @@
 (** {1:combinators Combinators} *)
 
 external id : 'a -> 'a = "%identity"
-(** [id] is the identity function. For any argument [x], [id x] is [x]. *)
+(** [id] is the identity function. For any argument [x], [id x] is [x].
+
+    example:
+    [List.init 3 id] is [[0; 1; 2]] *)
 
 val const : 'a -> (_ -> 'a)
 (** [const c] is a function that always returns the value [c]. For any
-    argument [x], [(const c) x] is [c]. *)
+    argument [x], [(const c) x] is [c].
+
+    example:
+    [List.init 3 (const 0)] is [[0; 0; 0]]
+
+    One important difference between [const x] and [fun _ -> x] in the above
+    example is that [x] is evaluated only once, not three times. *)
```

Comment:
Nitpick: with `x` I think of a variable (or maybe a value), not an expression, so evaluating it several times does not really make sense. To make this sentence precise we would ideally have a convention for metavariables denoting expressions, but we don't have one. I would write either

- `difference between [const foo] and [fun x -> foo] is that [foo] is evaluated only once`, or
- `difference between [const ...] and [fun x -> ...] is that [...] is evaluated only once`.

(I removed "in the above example": talking about this in the context of the example above is not great, because `0` is a value.)

Maybe we could use a specific example: `List.init 3 (Fun.const (Random.bool ())` draws a single boolean and uses it three times, whereas `List.init 3 (fun _ -> Random.bool ())` draws three random booleans.

Reply:
Thanks for the comments! Will create a PR with these examples with your comments and the other example PRs taken into account.
I like the ellipsis personally, and I considered using side effects to show how evaluation matters but I feel like side effects may make reasoning about the examples less straightforward.